pkg/gds/backup.go

```diff
@@ -78,6 +83,14 @@ backups:
 				log.Debug().Int("kept", s.conf.Backup.Keep).Int("removed", removed).Msg("backup directory cleaned up")
 			}
 		}
+
+		if stop != nil {
+			// Stop if someone wrote to the stop channel
+			for range stop {
+				log.Info().Msg("backup manager received stop signal")
+				return
+			}
+		}
```

Comment:
The backup manager keeps running once it's started, so I added some synchronization here so we can start/stop it within the context of a single test.